broadinstitute/pearl
# AR-73 site content data model

This adds the SiteContent data model as an improved take on the demo's content model.  Eventually, we may want to attach SiteContents to studies too, but this will suffice for now.  

Notably, this was actually faster to create than doing the same with Hibernate.  Although Hibernate created most of the schema faster, it took me a looong time to arrive at the magic incantation
```
    @Lob
    @Column
    @Type(type = "org.hibernate.type.BinaryType")
    @Getter @Setter
    private byte[] data;
```
to get hibernate to store binary data -- there were a lot of conflicting documentation and StackOverflow answers that depended on specific versions of Hibernate and Postgres (and Spring boot required us to use an older version of Hibernate that only had drivers for Postgres 10, while we use Postgres 15). 

TO TEST:
1. nuke/reset database
2. run PopulateCliApp with args 'portal' and 'portals/ourhealth/portal.json'
3. confirm database populates, notably the site_image and html_section tables
